tools/lint/black.py

```diff
@@ -0,0 +1,154 @@
+        black_reformats = int(reformats_matches[0])
+    # Look for a pattern like: "Y files would fail to reformat"
+    fails_matches = re.findall(r"(\d+)(?=\s+fil.+ would fail to reformat)", output)
+    if fails_matches:
+        black_fails = int(fails_matches[0])
+    logging.debug(
+        f"Parsed Black output: {black_reformats} files would be reformatted, {black_fails} files would fail"
+    )
+    return black_reformats, black_fails
+
+
+def generate_markdown_report(
+    black_reformats: int, black_fails: int, log_file: str
+) -> str:
+    """Generate a Markdown report section based on Black results and log file."""
+    report_lines = []
+    if black_reformats > 0 or black_fails > 0:
+        report_lines.append(
+            f"<details><summary>:pencil2: Black would reformat {black_reformats} file(s) and {black_fails} file(s) would fail to reformat</summary>"
+        )
+    else:
+        report_lines.append(
+            "<details><summary>:heavy_check_mark: Black would reformat no file</summary>"
+        )
+    report_lines.append("")
+    report_lines.append("```")
+    with open(log_file, "r", encoding="utf-8") as f:
+        report_lines.append(f.read())
+    report_lines.append("```")
+    report_lines.append("</details>")
+    report_lines.append("")
+    return "\n".join(report_lines)
+
+
+def emit_black_problem_matchers(log_path: str):
+    """Emit GitHub Actions problem matcher commands for Black output."""
+    if os.path.isfile(log_path):
+        RUNNER_WORKSPACE = os.getenv("RUNNER_WORKSPACE")
+        print(f"::add-matcher::{RUNNER_WORKSPACE}/FreeCAD/.github/problemMatcher/blackWarning.json")
+        with open(log_path, "r", encoding="utf-8") as f:
+            sys.stdout.write(f.read())
+        print("::remove-matcher owner=black-warning::")
+
+
+def append_report(report_file: str, report: str):
+    """Append the generated Markdown report to the specified report file."""
+    logging.info(f"Appending report to {report_file}")
+    with open(report_file, "a", encoding="utf-8") as f:
+        f.write(report + "\n")
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run Black in check mode on Python files and append a Markdown report."
+    )
+    parser.add_argument(
+        "--files",
+        required=True,
+        help="A space-separated list or glob pattern of Python files to check.",
+    )
+    parser.add_argument(
+        "--line-length",
+        default="100",
+        help="Maximum line length to use with Black (default: 100).",
+    )
+    parser.add_argument(
+        "--log-dir", required=True, help="Directory where log files will be written."
+    )
+    parser.add_argument(
+        "--report-file",
+        required=True,
+        help="Path to the Markdown report file to append results.",
+    )
+    parser.add_argument("--verbose", action="store_true", help="Enable verbose output.")
+    args = parser.parse_args()
+    setup_logger(args.verbose)
```

Comment:
Wouldn't it be better to pass all arguments directly to black?

Reply:
@tritao yes, forwarding all unregistered arguments - or all arguments after `--` like in CMake

Reply:
Yeah, then I'd prefer the same behaviour for all. I just thought that passing all unknown parameters would make coding easier : not including all the ones individually. Anyways, not so important to me.